mojo/stdlib/src/utils/numerics.mojo

```diff
@@ -509,64 +509,36 @@ fn nan[type: DType]() -> Scalar[type]:
 
     @parameter
     if type is DType.float8_e5m2:
-        return rebind[__mlir_type[`!pop.scalar<`, type.value, `>`]](
-            __mlir_op.`kgen.param.constant`[
-                _type = __mlir_type[`!pop.scalar<f8e5m2>`],
-                value = __mlir_attr[`#pop.simd<"nan"> : !pop.scalar<f8e5m2>`],
-            ]()
+        return rebind[Scalar[type]](
+            __mlir_attr.`#pop.simd<"nan"> : !pop.scalar<f8e5m2>`,
         )
```

Comment:
I wonder if you could further collapse all of these param ifs with something like:

```
  return rebind[Scalar[type]](
            __mlir_attr[`#pop.simd<"nan"> : `, Scalar[type]._mlir_type]
        )
```

This is how I collapsed similar code in `SIMD.init(self, value: FloatLiteral)`